geomesa-core/src/main/scala/org/locationtech/geomesa/core/iterators/DensityIterator.scala

```diff
@@ -294,5 +219,12 @@ object DensityIterator extends Logging {
     }
     table
   }
+}
 
+case class DensityIteratorResult(geometry: Geometry,
+                                 densityGrid: SparseMatrix = HashBasedTable.create[Double, Double, Long]()) extends Result {
+  override def addToFeature(featureBuilder: SimpleFeatureBuilder): Unit = {
+    featureBuilder.add(DensityIterator.encodeSparseMatrix(densityGrid))
+    featureBuilder.add(geometry)
```

Comment:
do we need to keep the geometry around here? maybe we could just use a static geom? that way we wouldn't have to keep copying the result up in handleKeyValue:107